.github/workflows/auto-pr-submodules.yml

```diff
@@ -0,0 +1,38 @@
+# This is a basic workflow to help you get started with Actions
+name: Update Submodules and Submit a PR
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push  only for the actions-test-branch branches
+  push:
+    branches:
+      - master
+    # Runs every day at midnight UTC
+    schedule:
+      - cron:  '0 0 * * *'
+
+jobs:
+  createPullRequest:
+    if: ${{ contains(github.ref, 'master') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
actions/checkout has options for getting submodules too. See https://github.com/actions/checkout/#usage.

It might make sense to suggest option `submodules: remote` to be added upstream.

Reply:
Wouldn't hurt I guess.

Reply:
`submodules: remote`?

Reply:
BTW We probably want a ["Treeless clone"](https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/#user-content-treeless-clones) otherwise the worker will run out of disk space?

Reply:
@mithro: Yeah, with the current setup a Github-hosted runner will definitely run out of disk space. But would a treeless clone help here tho (I'm trying it out locally right now)?

Reply:
@mithro, @umarcor: I'm afraid a treeless clone won't cut it (unless I'm doing it wrong). We will need a self-hosted runner to do this. However, since this workflow will only run on internal pushes, there are no security concerns regarding the protection of the self-hosted runner against pull-requests. In other words, this can be any machine.

After running the script to update submodules, after `git clone --filter=tree:0 ....`:
```bash
du -sh skywater-pdk/
53G	skywater-pdk/
```